delivery_roulier/models/stock_picking.py

```
@@ -0,0 +1,249 @@
+        # base_delivery_carrier_label module ensure packages is available
+        self.number_of_packages = len(packages)
+        self.carrier_tracking_ref = True  # display button in view
+        return packages._generate_labels(self)
+
+    # Default implementations of _roulier_*()
+    def _roulier_get_auth(self, account, package=None):
+        """Login/password of the carrier account.
+
+        Returns:
+            a dict with login and password keys
+        """
+        auth = {
+            'login': account.account,
+            'password': account.password,
+            'isTest': not self.carrier_id.prod_environment,
+        }
+        return auth
+
+    def _roulier_get_account(self, package=None):
+        """Return an 'account'.
+
+        By default, the first account encoutered for this type.
+        Depending on your case, you may store it on the picking or
+        compute it from your business rules.
+
+        Accounts are resolved at runtime (can be != for dev/prod)
+        """
+        self.ensure_one()
+        domain = [
+            ("name", "=", self.carrier_id.delivery_type),
+            "|",
+            ("company_id", "=", self.company_id.id),
+            ("company_id", "=", False)
+        ]
+        account = self.env["carrier.account"].search(domain, limit=1)
+        if not account:
+            raise UserError(
+                _("No account available with name '%s' "
+                  "for this carrier" % self.carrier_id.delivery_type))
+        return account
+
+    def _roulier_get_sender(self, package=None):
+        """Sender of the picking (for the label).
+
+        Return:
+            (res.partner)
+        """
+        return self.company_id.partner_id
+
+    def _roulier_get_label_format(self, account):
+        """format of the label asked for carrier
+
+        Return:
+            label format (string)
+        """
+        return getattr(account, '%s_file_format' % self.delivery_type, None)
```

why don't you fall back to `account.file_format`?

---

I am not sure the `file_format` will be relevant for any specific carrier.
Maybe we could fallback on this, but we hide it in the view and I don't really see cases where the carrier will have the same choices.
For me this `file_type` may not be relevant on the `carrier.account`, because not generic enough.